php-checks/src/main/java/org/sonar/php/checks/ParameterSequenceCheck.java

```diff
@@ -21,62 +21,66 @@
 import org.sonar.plugins.php.api.tree.Tree;
 import org.sonar.plugins.php.api.tree.Tree.Kind;
-import org.sonar.plugins.php.api.tree.declaration.FunctionTree;
+import org.sonar.plugins.php.api.tree.declaration.ClassDeclarationTree;
 import org.sonar.plugins.php.api.tree.declaration.NamespaceNameTree;
+import org.sonar.plugins.php.api.tree.expression.AnonymousClassTree;
 import org.sonar.plugins.php.api.tree.expression.ExpressionTree;
 import org.sonar.plugins.php.api.tree.expression.FunctionCallTree;
 import org.sonar.plugins.php.api.tree.expression.MemberAccessTree;
 import org.sonar.plugins.php.api.tree.expression.NameIdentifierTree;
 import org.sonar.plugins.php.api.tree.expression.VariableIdentifierTree;
 import org.sonar.plugins.php.api.visitors.PHPVisitorCheck;
 
-import javax.annotation.CheckForNull;
+import static org.sonar.plugins.php.api.symbols.QualifiedName.qualifiedName;
 
 @Rule(key = "S2234")
 public class ParameterSequenceCheck extends PHPVisitorCheck {
 
   private static final String MESSAGE = "Parameters to \"%s\" have the same names but not the same order as the method arguments.";
   private static final String SECONDARY_MESSAGE = "Implementation of the parameters sequence.";
 
+  private ClassSymbol currentClassSymbol;
+  private boolean isInAnonymousClass = false;
+
+  @Override
+  public void visitClassDeclaration(ClassDeclarationTree tree) {
+    currentClassSymbol = Symbols.get(tree);
+    super.visitClassDeclaration(tree);
+    currentClassSymbol = null;
+  }
+
   @Override
   public void visitFunctionCall(FunctionCallTree tree) {
     // only check method and function calls expect constructors with more than 1 argument
     if (!tree.getParent().is(Kind.NEW_EXPRESSION) && tree.arguments().size() > 1) {
-      if (tree.callee().is(Kind.NAMESPACE_NAME)) {
-        checkFunctionCall(tree, (NamespaceNameTree) tree.callee());
-      } else {
-        checkMethodCall(tree);
-      }
+      getDeclarationSymbol(tree).ifPresent(d -> checkFunctionCall(tree, d));
     }
 
     super.visitFunctionCall(tree);
   }
 
-  private void checkMethodCall(FunctionCallTree tree) {
-    Symbol symbol = getDeclarationSymbol(tree);
-    if (symbol != null && symbol.declaration() != null && symbol.is(Symbol.Kind.FUNCTION)) {
-      List<String> parameters = ((FunctionTree) symbol.declaration().getParent()).parameters().parameters().stream()
-        .map(e -> e.variableIdentifier().text())
-        .collect(Collectors.toList());
-
-      if (isWrongParameterSequence(tree, parameters)) {
-        newIssue(tree, String.format(MESSAGE, symbol.declaration().text()))
-          .secondary(symbol.declaration(), SECONDARY_MESSAGE);
-      }
-    }
+  @Override
+  public void visitAnonymousClass(AnonymousClassTree tree) {
+    boolean isInAnonymousClassBack = isInAnonymousClass;
+    isInAnonymousClass = true;
+    super.visitAnonymousClass(tree);
+    isInAnonymousClass = isInAnonymousClassBack;
```

Comment:
It is possible to have a class declaration inside an anonymous class? Would this code work for such a case?

Reply:
It wouldn't have surprise me if this would have been possible in PHP :) but it doesn't seem so according to the grammar.